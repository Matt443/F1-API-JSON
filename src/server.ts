import { getDriverData } from "./scraper/driver-data";
import { getTeamsData } from "./scraper/team-data";
import { getDriverStandings } from "./scraper/driver-standings";
import { getConstructorStandings } from './scraper/constructors-standings';
import { getWorldChampions } from './scraper/world-champions'

export {
    getDriverData,
    getTeamsData,
    getDriverStandings,
    getConstructorStandings,
    getWorldChampions
}