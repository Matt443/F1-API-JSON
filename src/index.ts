import { getRaceResults } from "./server";

async function test() {
    await getRaceResults();
    console.log();
}
test();
