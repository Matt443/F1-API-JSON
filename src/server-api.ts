import express from "express";
import cors from "cors";
import helmet from "helmet";
import dotenv from "dotenv";

import router from "./endpoints-api/routes-api";
import { getFullRaceResults, getRaceResults } from "./server";

dotenv.config();

const app = express();

app.use(cors());
app.use(helmet());

app.use("/", router);

app.listen(process.env.PORT || 4005, () => {
    console.log(`server is running on ${process.env.PORT || 4005}`);
}).on("error", (error) => {
    throw new Error(error.message);
});
