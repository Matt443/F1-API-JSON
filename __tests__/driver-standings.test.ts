import { getDriverStandings } from "../src/server";

test("driver standings", async () => {
    expect(await getDriverStandings()).toMatchSnapshot();
});
