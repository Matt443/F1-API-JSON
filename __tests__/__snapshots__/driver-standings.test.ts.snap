// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`driver standings 1`] = `
[
  {
    "driver": "Max Verstappen",
    "nationality": "NED",
    "points": 454,
    "position": 1,
    "team": "Red Bull Racing RBPT",
  },
  {
    "driver": "Charles Leclerc",
    "nationality": "MON",
    "points": 308,
    "position": 2,
    "team": "Ferrari",
  },
  {
    "driver": "Sergio Perez",
    "nationality": "MEX",
    "points": 305,
    "position": 3,
    "team": "Red Bull Racing RBPT",
  },
  {
    "driver": "George Russell",
    "nationality": "GBR",
    "points": 275,
    "position": 4,
    "team": "Mercedes",
  },
  {
    "driver": "Carlos Sainz",
    "nationality": "ESP",
    "points": 246,
    "position": 5,
    "team": "Ferrari",
  },
  {
    "driver": "Lewis Hamilton",
    "nationality": "GBR",
    "points": 240,
    "position": 6,
    "team": "Mercedes",
  },
  {
    "driver": "Lando Norris",
    "nationality": "GBR",
    "points": 122,
    "position": 7,
    "team": "McLaren Mercedes",
  },
  {
    "driver": "Esteban Ocon",
    "nationality": "FRA",
    "points": 92,
    "position": 8,
    "team": "Alpine Renault",
  },
  {
    "driver": "Fernando Alonso",
    "nationality": "ESP",
    "points": 81,
    "position": 9,
    "team": "Alpine Renault",
  },
  {
    "driver": "Valtteri Bottas",
    "nationality": "FIN",
    "points": 49,
    "position": 10,
    "team": "Alfa Romeo Ferrari",
  },
  {
    "driver": "Daniel Ricciardo",
    "nationality": "AUS",
    "points": 37,
    "position": 11,
    "team": "McLaren Mercedes",
  },
  {
    "driver": "Sebastian Vettel",
    "nationality": "GER",
    "points": 37,
    "position": 12,
    "team": "Aston Martin Aramco Mercedes",
  },
  {
    "driver": "Kevin Magnussen",
    "nationality": "DEN",
    "points": 25,
    "position": 13,
    "team": "Haas Ferrari",
  },
  {
    "driver": "Pierre Gasly",
    "nationality": "FRA",
    "points": 23,
    "position": 14,
    "team": "AlphaTauri RBPT",
  },
  {
    "driver": "Lance Stroll",
    "nationality": "CAN",
    "points": 18,
    "position": 15,
    "team": "Aston Martin Aramco Mercedes",
  },
  {
    "driver": "Mick Schumacher",
    "nationality": "GER",
    "points": 12,
    "position": 16,
    "team": "Haas Ferrari",
  },
  {
    "driver": "Yuki Tsunoda",
    "nationality": "JPN",
    "points": 12,
    "position": 17,
    "team": "AlphaTauri RBPT",
  },
  {
    "driver": "Zhou Guanyu",
    "nationality": "CHN",
    "points": 6,
    "position": 18,
    "team": "Alfa Romeo Ferrari",
  },
  {
    "driver": "Alexander Albon",
    "nationality": "THA",
    "points": 4,
    "position": 19,
    "team": "Williams Mercedes",
  },
  {
    "driver": "Nicholas Latifi",
    "nationality": "CAN",
    "points": 2,
    "position": 20,
    "team": "Williams Mercedes",
  },
  {
    "driver": "Nyck De Vries",
    "nationality": "NED",
    "points": 2,
    "position": 21,
    "team": "Williams Mercedes",
  },
  {
    "driver": "Nico Hulkenberg",
    "nationality": "GER",
    "points": 0,
    "position": 22,
    "team": "Aston Martin Aramco Mercedes",
  },
]
`;
